transformers/struct_test.go

```diff
@@ -480,3 +480,153 @@ func TestTableFromGoStruct(t *testing.T) {
+			testStruct: struct {
+				Tags map[string]string `json:"tags"`
+			}{},
+			want: map[string]string{
+				"tags": "map<utf8, utf8, items_nullable>",
+			},
+		},
+		{
+			name: "simple array",
+			testStruct: struct {
+				Items []struct {
+					Name string `json:"name"`
+				} `json:"items"`
+			}{},
+			want: map[string]string{
+				"items": `list<{"name":"utf8"}, items_nullable>`,
+			},
+		},
+		{
+			name: "simple struct",
+			testStruct: struct {
+				Item struct {
+					Name string `json:"name"`
+				} `json:"item"`
+			}{},
+			want: map[string]string{
+				"item": `{"name":"utf8"}`,
+			},
+		},
+		{
+			name: "complex struct",
+			testStruct: struct {
+				Item struct {
+					Name         string            `json:"name"`
+					Tags         map[string]string `json:"tags"`
+					FlatItems    []string          `json:"flat_items"`
+					ComplexItems []struct {
+						Name string `json:"name"`
+					} `json:"complex_items"`
+				} `json:"item"`
+			}{},
+			want: map[string]string{
+				"item": `{"complex_items":"list<{\"name\":\"utf8\"}, items_nullable>","flat_items":"list<item: utf8, nullable>","name":"utf8","tags":"map<utf8, utf8, items_nullable>"}`,
+			},
+		},
+		{
+			name: "multiple json columns",
+			testStruct: struct {
+				Tags map[string]string `json:"tags"`
+				Item struct {
+					Name         string            `json:"name"`
+					Tags         map[string]string `json:"tags"`
+					FlatItems    []string          `json:"flat_items"`
+					ComplexItems []struct {
+						Name string `json:"name"`
+					} `json:"complex_items"`
+				} `json:"item"`
+			}{},
+			want: map[string]string{
+				"item": `{"complex_items":"list<{\"name\":\"utf8\"}, items_nullable>","flat_items":"list<item: utf8, nullable>","name":"utf8","tags":"map<utf8, utf8, items_nullable>"}`,
+			},
+		},
+		{
+			name: "handles any type in struct",
+			testStruct: struct {
+				Item struct {
+					Name   string `json:"name"`
+					Object any    `json:"object"`
+				} `json:"item"`
+			}{},
+			want: map[string]string{
+				"item": `{"name":"utf8","object":"any"}`,
+			},
+		},
+		{
+			name: "handles map from string to any",
+			testStruct: struct {
+				Tags map[string]any `json:"tags"`
+			}{},
+			want: map[string]string{
+				"tags": "map<utf8, any, items_nullable>",
+			},
+		},
+		{
+			name: "handles array of any",
+			testStruct: struct {
+				Items []any `json:"items"`
+			}{},
+			want: map[string]string{
+				"items": `list<any, items_nullable>`,
+			},
+		},
+		{
+			name: "stops at the default depth of 5",
+			testStruct: struct {
+				Level0 struct {
+					Level1 struct {
+						Level2 struct {
+							Level3 struct {
+								Level4 struct {
+									Level5 struct {
+										Level6 struct {
+											Name string `json:"name"`
+										} `json:"level6"`
+									} `json:"level5"`
+								} `json:"level4"`
+							} `json:"level3"`
+						} `json:"level2"`
+					} `json:"level1"`
+				} `json:"level0"`
+			}{},
+			want: map[string]string{
```

Comment:
Each level is a JSON encoded schema hence the quotes. Still looking if this can be improved